extension/json/json_functions/json_pretty_print.cpp

```diff
@@ -0,0 +1,32 @@
+#include "json_executors.hpp"
+
+namespace duckdb {
+
+//! Pretty Print a given JSON Document
+string_t PrettyPrint(yyjson_val *val, yyjson_alc *alc, Vector &result) {
+	D_ASSERT(alc);
+	idx_t len;
+	auto data =
+	    yyjson_val_write_opts(val, JSONCommon::WRITE_PRETTY_FLAG, alc, reinterpret_cast<size_t *>(&len), nullptr);
+	return string_t(data, len);
```

Comment:
I'm not seeing StringVector::AddString anywhere, which worries me
`string_t` does not take ownership of the memory.

Which is why we often use StringVector::AddString, which copies it to the StringHeap that's owned by the Vector, guaranteeing that the string_t we return will point to memory that lives as long as the Vector that's holding it

Reply:
This is also used in `json_serialize_plan` in the file that I referenced